### PR TITLE
DROID-1686 Tech | Target SDK 34

### PR DESCRIPTION
**Pull request by konstantiniiv**
- Target SDK version updated
- Libraries updated
- SafeArgs library deleted
- Compose navigation animation migrated ([Migration Guide](https://google.github.io/accompanist/navigation-animation/#migration))
- Launch on custom MW version tested

### CODE REVIEW

File: app/src/main/java/com/anytypeio/anytype/ui/onboarding/OnboardingFragment.kt [hunk 1]
```diff
@@ -413,7 +408,7 @@ class OnboardingFragment : Fragment() {
             vm.sideEffects.collect { effect ->
                 when (effect) {
                     is OnboardingMnemonicLoginViewModel.SideEffect.Exit -> {
-                        val lastDestination = navController.backQueue.lastOrNull()
+                        val lastDestination = navController.currentBackStackEntry
```
Comment (konstantiniiv) [inline app/src/main/java/com/anytypeio/anytype/ui/onboarding/OnboardingFragment.kt:411]: https://android.googlesource.com/platform/frameworks/support/+/9bf172dde12c29f1187576cab4e7ce2d0989099a%5E%21/

File: app/src/main/java/com/anytypeio/anytype/ui/onboarding/OnboardingFragment.kt [hunk 2]
```diff
@@ -139,10 +136,9 @@ class OnboardingFragment : Fragment() {
     }
 
     @Composable
-    @OptIn(ExperimentalAnimationApi::class)
     private fun OnboardingScreen() {
         MaterialTheme {
-            val navController = rememberAnimatedNavController()
+            val navController = rememberNavController()
```
Comment (uburoiubu) [inline app/src/main/java/com/anytypeio/anytype/ui/onboarding/OnboardingFragment.kt:141]: does slide animation still work?

File: core-utils/src/main/java/com/anytypeio/anytype/core_utils/OnSwipeListener.kt [hunk 3]
```diff
@@ -34,8 +34,8 @@ abstract class OnSwipeListener : SimpleOnGestureListener() {
         //  C => it's a DOWN swipe
         //  D => it's a LEFT swipe
         //
-        val x1 = e1.x
-        val y1 = e1.y
+        val x1 = e1?.x ?: 0f
```
Comment (uburoiubu) [inline core-utils/src/main/java/com/anytypeio/anytype/core_utils/OnSwipeListener.kt:37]: please if UX behavior related to this code is not broken

**Review by uburoiubu — APPROVED** [review]

File: core-utils/src/main/java/com/anytypeio/anytype/core_utils/OnSwipeListener.kt (hunk 3 above)
Comment (konstantiniiv) [reply core-utils/src/main/java/com/anytypeio/anytype/core_utils/OnSwipeListener.kt:37]: I checked everything works, but the view this listener is working with will be removed very soon

File: app/src/main/java/com/anytypeio/anytype/ui/onboarding/OnboardingFragment.kt (hunk 2 above)
Comment (konstantiniiv) [reply app/src/main/java/com/anytypeio/anytype/ui/onboarding/OnboardingFragment.kt:141]: checked, everything is working as it should be

**Pull request merged**